cosmology.c

```diff
@@ -36,8 +36,19 @@ void init_cosmology(Cosmology * CP_in, const double TimeBegin)
     init_omega_nu(&CP->ONu, CP->MNu, TimeBegin, CP->HubbleParam, CP->CMBTemperature);
     /* Neutrinos will be included in Omega0, if massive.
      * This ensures that OmegaCDM contains only non-relativistic species.*/
-    if(CP->MNu[0] + CP->MNu[1] + CP->MNu[2] > 0)
+    if(CP->MNu[0] + CP->MNu[1] + CP->MNu[2] > 0) {
         CP->OmegaCDM -= get_omega_nu(&CP->ONu, 1);
+        /* Warn if we have a large enough box that neutrinos are not free-streaming on the largest scales.
+         * Free-streaming scale is ~500 Mpc at z=99 for M_nu = 0.3 eV.
+         * FIXME: Implement scale dependent growth correction, by differentiating transfer functions.*/
+        /*Neutrino thermal velocity in cm/s*/
+        double v_th = BOLEVK*(TNUCMB*CP->CMBTemperature)/(CP->MNu[0] + CP->MNu[1] + CP->MNu[2])*3. / TimeBegin * C;
+        /*Neutrino free-streaming length from Lesgourgues & Pastor in (comoving) cm/h*/
+        double nufs = 2 * M_PI * sqrt(2/3.) * v_th / (hubble_function(TimeBegin)/CP->Hubble * HUBBLE)/TimeBegin;
+        /*Box should be in comoving cm/h: we ask for 1/4 of the box because sample variance will matter on larger scales.*/
+        if(nufs < box/4.)
```

Comment:
Shall we lift the logic out of cosmology by adding a function that gets the nu free streming scale of any redshift / a?

Reply:
This is only temporary - we will implement scale-dependent growth functions (also for the baryons)